mypy failure: unused error code in numpy2_compat.py

after merging with main just now, #4996 is failing the type-check test with an unrelated failure ([link](https://github.com/yt-project/yt/actions/runs/12934059343/job/36074148625?pr=4996)):

```
yt/_maintenance/numpy2_compat.py:9: error: Unused "type: ignore" comment  [unused-ignore]
yt/_maintenance/numpy2_compat.py:9: error: Incompatible types in assignment (expression has type "Callable[[_Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | builtins.bool | int | float | complex | str | bytes | _NestedSequence[builtins.bool | int | float | complex | str | bytes], _Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | builtins.bool | int | float | complex | str | bytes | _NestedSequence[builtins.bool | int | float | complex | str | bytes] | None, float, int], generic[Any] | ndarray[tuple[int, ...], dtype[generic[Any]]]]", variable has type overloaded function)  [assignment]
yt/_maintenance/numpy2_compat.py:9: note: Error code "assignment" not covered by "type: ignore" comment
Found 2 errors in 1 file (checked 516 source files)
```

--- a/yt/_maintenance/numpy2_compat.py
+++ b/yt/_maintenance/numpy2_compat.py
@@ -6,4 +6,4 @@ if hasattr(np, "trapezoid"):
     # np.trapz is deprecated in numpy 2.0 in favor of np.trapezoid
     trapezoid = np.trapezoid
 else:
-    trapezoid = np.trapz  # type: ignore [attr-defined] # noqa: NPY201
+    trapezoid = np.trapz  # type: ignore[assignment] # noqa: NPY201

